### PR TITLE
feature(CacheItemSelector): add a condition in 'forget' method to be …

### PR DESCRIPTION
**Pull request by jbidad**
add a condition in 'forget' method to be able to forget by hashes(keys) also

### CODE REVIEW

File: src/CacheItemSelector/CacheItemSelector.php [hunk 1]
```diff
@@ -35,6 +36,14 @@ public function forget(): void
     {
         collect($this->urls)
             ->map(function ($uri) {
+                if (
+                    is_string($uri) &&
```
Comment (freekmurze) [inline src/CacheItemSelector/CacheItemSelector.php:40]: I don't understand what this does. Could you refactor this to a method with a name that explains what it does? Don't use `&&` but early returns in that method.

**Review by freekmurze — CHANGES_REQUESTED** [review]
Could you also add a test for this?

File: src/CacheItemSelector/CacheItemSelector.php (hunk 1 above)
Comment (jbidad) [reply src/CacheItemSelector/CacheItemSelector.php]: thank you for your comment, I add 'isUriHashed' to explain that more clear
@freekmurze

**Comment by freekmurze** [comment]
Could you add tests for the new functionality?

**Pull request closed without merging**